### PR TITLE
fix(TDP-5595): fetch more doesn't work most of the time

### PR DESCRIPTION
**Pull request by skermabon**
* Move unit tests about aggregation that not really unit tests but integration tests to integration tests

**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-5595

**Please check if the PR fulfills these requirements**
- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [ ] The new code does not introduce new technical issues (sonar / eslint)
- [ ] Functional tests have been performed
- [ ] Docker configuration files for config-std or config-cloud profiles are impacted

**Please check the browsers you've tested on**
- [ ] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [ ] No, and no need to (backend changes only)

### CODE REVIEW

File: dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/AggregateStep.java [hunk 1]
```diff
@@ -54,12 +56,60 @@ public void applyAnAggregation(String preparationName, DataTable dataTable) thro
...
+        }
+
+        Aggregate aggregate = createAggregate(params);
+
+        Response response = api.applyAggragate(aggregate);
+        response.then().statusCode(value);
+    }
+
     @Then("^The aggregate result with the operator \"(.*)\" is :$")
     public void testAggregate(String operator, DataTable dataTable) throws Exception {
```
Comment (yacarof) [inline dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/AggregateStep.java]: I would rather prefer something like :
"^The aggregation "(.)" results are :$")

File: dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/AggregateStep.java [hunk 2]
```diff
@@ -54,12 +56,60 @@ public void applyAnAggregation(String preparationName, DataTable dataTable) thro
...
         Map<String, String> params = dataTable.asMap(String.class, String.class);
 
         List<AggregateResult> aggregateResults = (List<AggregateResult>) (context.getObject("aggregate"));
-        assertEquals(aggregateResults, toAggregateResult(params, operator));
+        assertEquals(toAggregateResult(params, operator), aggregateResults);
+    }
+
+    @When("^I apply an aggregation on the dataSet \"(.*)\" with parameters :$")
+    public void applyAnAggregationOnDataSet(String dataSetName, DataTable dataTable) throws Exception {
```
Comment (yacarof) [inline dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/AggregateStep.java]: I would rather prefer something like :
"^I create the aggregation "(.)" on the dataset "(.)" with the following parameters :$")

File: dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/AggregateStep.java [hunk 3]
```diff
@@ -39,8 +41,8 @@
  */
 public class AggregateStep extends DataPrepStep {
 
-    @When("^I apply an aggregation preparation \"(.*)\" with parameters :$")
-    public void applyAnAggregation(String preparationName, DataTable dataTable) throws Exception {
+    @When("^I apply an aggregation on the preparation \"(.*)\" with parameters :$")
```
[the remark this answers is not in the document]
Comment (skermabon) [reply dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/AggregateStep.java]: ok for me. Changes done

File: dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/AggregateStep.java (hunk 1 above)
Comment (skermabon) [reply dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/AggregateStep.java]: ok for me. Changes done

File: dataprep-test-api/src/test/java/org/talend/dataprep/qa/dto/PreparationDetails.java [hunk 4]
```diff
@@ -45,4 +45,14 @@ public void updateActionIds() {
 
         public String dataSetNbRow;
     }
+
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    public String getHead() {
```
[the remark this answers is not in the document]
Comment (skermabon) [reply dataprep-test-api/src/test/java/org/talend/dataprep/qa/dto/PreparationDetails.java]: It's strange to use ExportUtil for an aggregation and not an export. No?

File: dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/AggregateStep.java [hunk 5]
```diff
@@ -49,17 +51,65 @@ public void applyAnAggregation(String preparationName, DataTable dataTable) thro
...
+        aggregationFailled(preparationName, null, dataTable, NOT_ACCEPTABLE.value());
+    }
+
+    @When("^I fail to apply an aggregation on preparation \"(.*)\" and dataSet \"(.*)\" with parameters :$")
+    public void applyAnAggregationOnNonExistingPreparationFailed(String preparationName, String dataSetName,
+            DataTable dataTable) throws Exception {
+        aggregationFailled(preparationName, dataSetName, dataTable, BAD_REQUEST.value());
+    }
+
+    private void aggregationFailled(String preparationName, String dataSetName, DataTable dataTable, int value)
```
Comment (tlnd-jbrun) [inline dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/AggregateStep.java]: Faillllllled ? :-)

File: dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/AggregateStep.java [hunk 6]
```diff
@@ -49,17 +51,65 @@ public void applyAnAggregation(String preparationName, DataTable dataTable) thro
...
+            throws Exception {
+        Map<String, String> params = new HashMap<>(dataTable.asMap(String.class, String.class));
+        if (preparationName != null) {
+            params.put("preparationId", context.getPreparationId(suffixName(preparationName)));
```
Comment (tlnd-jbrun) [inline dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/AggregateStep.java]: could be good to use constant

File: dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/AggregateStep.java [hunk 7]
```diff
@@ -49,17 +51,65 @@ public void applyAnAggregation(String preparationName, DataTable dataTable) thro
...
+        }
+        if (dataSetName != null) {
+            params.put("dataSetId", context.getDatasetId(suffixName(dataSetName)));
```
Comment (tlnd-jbrun) [inline dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/AggregateStep.java]: same thing

File: dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/AggregateStep.java [hunk 8]
```diff
@@ -49,17 +51,65 @@ public void applyAnAggregation(String preparationName, DataTable dataTable) thro
...
+        context.storeObject(suffixName(aggregationName),
                 objectMapper.readValue(response.body().print(), new TypeReference<List<AggregateResult>>() {
                 }));
     }
 
-    @Then("^The aggregate result with the operator \"(.*)\" is :$")
-    public void testAggregate(String operator, DataTable dataTable) throws Exception {
+    @When("^I fail to apply an aggregation preparation \"(.*)\" with parameters :$")
+    public void applyAnAggregationOnPreparationFailed(String preparationName, DataTable dataTable) throws Exception {
+        aggregationFailled(preparationName, null, dataTable, BAD_REQUEST.value());
```
Comment (tlnd-jbrun) [inline dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/AggregateStep.java]: Failled

File: dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/AggregateStep.java [hunk 9]
```diff
@@ -49,17 +51,65 @@ public void applyAnAggregation(String preparationName, DataTable dataTable) thro
...
+    }
+
+    @When("^I fail to apply an aggregation on non existing preparation \"(.*)\" with parameters :$")
+    public void applyAnAggregationOnNonExistingPreparationFailed(String preparationName, DataTable dataTable)
+            throws Exception {
+        aggregationFailled(preparationName, null, dataTable, NOT_ACCEPTABLE.value());
```
Comment (tlnd-jbrun) [inline dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/AggregateStep.java]: Failllled

File: dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/AggregateStep.java [hunk 10]
```diff
@@ -49,17 +51,65 @@ public void applyAnAggregation(String preparationName, DataTable dataTable) thro
...
+    }
+
+    @When("^I fail to apply an aggregation on preparation \"(.*)\" and dataSet \"(.*)\" with parameters :$")
+    public void applyAnAggregationOnNonExistingPreparationFailed(String preparationName, String dataSetName,
+            DataTable dataTable) throws Exception {
+        aggregationFailled(preparationName, dataSetName, dataTable, BAD_REQUEST.value());
```
Comment (tlnd-jbrun) [inline dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/AggregateStep.java]: Faillled

File: dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/AggregateStep.java [hunk 11]
```diff
@@ -49,17 +51,65 @@ public void applyAnAggregation(String preparationName, DataTable dataTable) thro
...
-        List<AggregateResult> aggregateResults = (List<AggregateResult>) (context.getObject("aggregate"));
-        assertEquals(aggregateResults, toAggregateResult(params, operator));
+        List<AggregateResult> aggregateResults = (List<AggregateResult>) (context.getObject(suffixName(aggregationName)));
+        assertEquals(toAggregateResult(params, operator), aggregateResults);
+    }
+
+    @When("^I apply an aggregation \"(.*)\" on the dataSet \"(.*)\" with parameters :$")
+    public void applyAnAggregationOnDataSet(String aggregationName, String dataSetName, DataTable dataTable) throws Exception {
+        Map<String, String> params = new HashMap<>(dataTable.asMap(String.class, String.class));
+        params.put("dataSetId", context.getDatasetId(suffixName(dataSetName)));
```
Comment (tlnd-jbrun) [inline dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/AggregateStep.java]: constant

File: dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/AggregateStep.java [hunk 12]
```diff
@@ -94,9 +144,18 @@ private Aggregate createAggregate(Map<String, String> params) {
         AggregateOperation aggregateOperation = new AggregateOperation(params.get("operator"), params.get("columnId"));
         aggregate.addOperation(aggregateOperation);
 
-        aggregate.preparationId = params.get("preparationId");
+        if (params.get("preparationId") != null) {
+            aggregate.preparationId = params.get("preparationId");
```
Comment (tlnd-jbrun) [inline dataprep-test-api/src/test/java/org/talend/dataprep/qa/step/AggregateStep.java]: constant

**Pull request merged**